src/js/megamenu.js

```diff
@@ -1,38 +1,45 @@
-const langs = require("./data/langs");
-const sections = require("./data/sites");
-const localize = require("./util/localize");
-
-document.addEventListener("DOMContentLoaded", function () {
+document.addEventListener("DOMContentLoaded", function() {
   const megamenu = document.getElementById("megamenu");
   if (!megamenu) {
     return;
   }
-  const content = sections
-    .map(function (section) {
-      const links = section.links
-        .map(function (link) {
-          return `<li>${link.icon} <a class="l10n" href="${link.url}" data-msg-id="${link.id}" data-url-id="${link.id}-url">${link.title}</a></li>`;
+  fetch('//universe.opensuse.org/api/v0/sites.json').then(languages_response => {
+    return languages_response.json();
+  }).then(languages => {
+    var lang = document.documentElement.lang;
+    if (!lang || !(lang in languages)) {
+      lang = 'en';
+    }
+
+    fetch(languages[lang]).then(sections_response => {
+      return sections_response.json();
+    }).then(sections => {
```

Comment:
Here it always sends two HTTP requests. How about directly fetch `/api/v0/sites-de.json`. If the language doesn't exist, let Apache/Nginx to redirect to `/api/v0/sites-en.json`.

Reply:
Hmm, the issue is that we don't know if the language on the server side is xx_YY or xx, so we would need to make two requests that way as well